base/shared/src/main/scala/scalaz/typeclass/BindClass.scala

```diff
@@ -2,7 +2,7 @@ package scalaz
 package typeclass
 
 trait BindClass[M[_]] extends Bind[M] with ApplyClass[M] {
-  final def bind: Bind[M] = this
+  final def bind: Bind[M] with this.type = this
```

Comment:
I would love to make this a wart or something checkable.

Someone make and approve a scalaz compiler plugin and it will be so.